src/helper/markdown/lineQueries.ts

```diff
@@ -0,0 +1,63 @@
+// Special Tokens
+enum MarkdownTokens {
+    H1 = '# ',
+    H2 = '## ',
+    H3 = '### ',
+    H4 = '#### ',
+    H5 = '##### ',
+    H6 = '###### ',
+}
+
+function isHeadingLevel1(line: string): boolean {
+    return line.startsWith(MarkdownTokens.H1);
+}
+
+function isHeadingLevel2(line: string): boolean {
+    return line.startsWith(MarkdownTokens.H2);
+}
+
+function isHeadingLevel3(line: string): boolean {
+    return line.startsWith(MarkdownTokens.H3);
+}
+
+function isHeadingLevel4(line: string): boolean {
+    return line.startsWith(MarkdownTokens.H4);
+}
+
+function isHeadingLevel5(line: string): boolean {
+    return line.startsWith(MarkdownTokens.H5);
+}
+
+function isHeadingLevel6(line: string): boolean {
+    return line.startsWith(MarkdownTokens.H6);
+}
+
+// Line formatter
+export function formatHtmlForMarkdownLine(line: string) {
+    let formattedLine = line;
+
+    if (isHeadingLevel1(line)) {
+        formattedLine = `<h1>${line.replace(MarkdownTokens.H1, '')}</h1>`;
+    }
+    else if (isHeadingLevel2(line)) {
```

Comment:
same as above

Reply:
In this scenario, it might be possible to use a regex as all heading tokens are similar. But I chose this way to make it a scalable soluton as other forms of token might be way different making it impossible for using regex. Please share with me an example if you have an idea

Reply:
please see above example I mentioned.